### PR TITLE
Ltd 3543 audit query responses

### PR DESCRIPTION
**Pull request by antroy-madetech**
### Aim

Adds an audit of the ECJU query response from an exporter.

[LTD-3543](https://uktrade.atlassian.net/browse/LTD-3543)

[LTD-3543]: https://uktrade.atlassian.net/browse/LTD-3543?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/cases/tests/test_case_ecju_queries.py [hunk 1]
```diff
@@ -281,6 +284,12 @@ def _test_exporter_responds_to_query(self, add_documents, query_type):
         response = response.json()["ecju_query"]
         self.assertEqual(response["response"], data["response"])
 
+        lu_advice_audit = Audit.objects.filter(verb=AuditType.ECJU_QUERY_RESPONSE)
```
Comment (saruniitr) [inline api/cases/tests/test_case_ecju_queries.py:287]: this is not specific to LU

File: api/cases/views/views.py [hunk 2]
```diff
@@ -547,7 +547,13 @@ def put(self, request, pk, ecju_pk):
...
             if "validate_only" not in request.data or not request.data["validate_only"]:
                 serializer.save()
-
+                audit_trail_service.create(
+                    actor=request.user,
+                    verb=AuditType.ECJU_QUERY_RESPONSE,
+                    action_object=serializer.instance,
+                    target=serializer.instance.case,
+                    payload={"ecju_response": data["response"]},
+                )
```
Comment (saruniitr) [inline api/cases/views/views.py:556]: this looks good.
Could you check if this shows up in "Latest action" on All cases page as we are expecting that it gets picked up there once this audit is created.

File: api/cases/tests/test_case_ecju_queries.py (hunk 1 above)
Comment (antroy-madetech) [reply api/cases/tests/test_case_ecju_queries.py:287]: Copy paste error. I will change

**Review by saruniitr — APPROVED** [review]

File: api/cases/tests/test_case_ecju_queries.py [hunk 3]
```diff
@@ -281,6 +284,12 @@ def _test_exporter_responds_to_query(self, add_documents, query_type):
         response = response.json()["ecju_query"]
         self.assertEqual(response["response"], data["response"])
 
+        query_response_audit = Audit.objects.filter(verb=AuditType.ECJU_QUERY_RESPONSE)
+        self.assertTrue(query_response_audit.exists())
+        audit_obj = query_response_audit.first()
+        audit_text = AuditSerializer(audit_obj).data["text"]
+        self.assertEqual(audit_text, " responded to an ECJU Query: Attached the requested documents.")
```
Comment (currycoder) [inline api/cases/tests/test_case_ecju_queries.py:291]: Is it worth also making sure that this audit object is attached to the case in question?

**Review by currycoder — APPROVED** [review]
Looks great, just one optional comment

**Pull request merged**